docs/buildDocu.ps1

```diff
@@ -0,0 +1,75 @@
+Param(
+	[Switch]$Deploy
+)
+
+$git_user = "CiBuildDocu"
+$git_email = "CiBuildDocu@wacore.com"
+$git_access_token = "4vngnALEVBWUWd4tuLHADxd5vF26geMyGw0dGjun494fq/W0sLiQAJxhXXYsdD39"
+$target_branch = "gh-pages"
+
+$docfxVersion = "2.24.0"
+$VisualStudioVersion = "15.0";
+$DotnetSDKVersion = "2.0.0";
+
+# Get dotnet paths
+$MSBuildExtensionsPath = "C:\Program Files\dotnet\sdk\" + $DotnetSDKVersion;
+$MSBuildSDKsPath = $MSBuildExtensionsPath + "\SDKs";
+
+# Get Visual Studio install path
+$VSINSTALLDIR =  $(Get-ItemProperty "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\SxS\VS7").$VisualStudioVersion;
+
+# Add Visual Studio environment variables
+$env:VisualStudioVersion = $VisualStudioVersion;
+$env:VSINSTALLDIR = $VSINSTALLDIR;
+
+# Add dotnet environment variables
+$env:MSBuildExtensionsPath = $MSBuildExtensionsPath;
+$env:MSBuildSDKsPath = $MSBuildSDKsPath;
+
+# Install docfx
+& nuget install docfx.console -Version $docfxVersion
+
+if($Deploy){
+	# Configuring git credentials
+	Write-Host "`n[Configuring git credentials]" -ForegroundColor Green
+	& git config --global credential.helper store
+	Add-Content "$env:USERPROFILE\.git-credentials" "https://$($git_access_token):x-oauth-basic@github.com`n"
+
+	& git config --global user.email "$($git_email)"
+	& git config --global user.name "$($git_user)"
+
+	# Checkout gh-pages
+	Write-Host "`n[Checkout $target_branch]" -ForegroundColor Green
+	git clone --quiet --no-checkout --branch=$target_branch https://github.com/HerbertBodner/webapp-core.git docs/_site
+}
+
+# Build our docs
+Write-Host "`n[Build our docs]" -ForegroundColor Green
+
+& .\docfx.console.$docfxVersion\tools\docfx docs/docfx.json
+
+if($Deploy){
+	git -C docs/_site status
+	$pendingChanges = git -C docs/_site status | select-string -pattern "Changes not staged for commit:","Untracked files:" -simplematch
+	if ($pendingChanges -ne $null) { 
+		# Committing changes
+		Write-host "`n[Committing changes]" -ForegroundColor Green
+		Write-host "`n[git -C docs/_site add -A]" -ForegroundColor Green
+		git -C docs/_site add -A
+		Write-host "`n[git -C docs/_site status]" -ForegroundColor Green
+		git -C docs/_site status
+		Write-host "`n[Committing changes]" -ForegroundColor Green
+		git -C docs/_site commit -m "static site regeneration"
+		# Pushing changes
+		Write-host "`n[Pushing changes]" -ForegroundColor Green
+		git -C docs/_site push origin $target_branch
+		Write-Host "`n[Success!]" -ForegroundColor Green
+	} 
+	else { 
+		write-host "`nNo changes to documentation" -ForegroundColor Yellow
+	}
+
+	# Remove git files in user profile (for local execution of buildDocu.ps)
```

Comment:
Whoa, I would never run this script locally because I don't want it to modify and then delete my .gitconfig and .git-credentials!
How about adding the credentials to the git url when pushing? like 
`git -C docs/_site push https://$git_access_token:x-oauth-basic@github.com/HerbertBodner/webapp-core.git $target_branch`

For the .gitconfig file, you could configure username and email after cloning, in the cloned repository.
```
& git config -C docs/_site --local user.email "$($git_email)"
& git config -C docs/_site --local user.name "$($git_user)"
```
This way you don't touch the global config and global credential store at all.